test/tests/functional/pbs_alps_request_tunables.py

```diff
@@ -0,0 +1,150 @@
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+import math
+
+
+@tags('cray')
+class TestCrayAlpsRequestTunables(TestFunctional):
+    """
+    Set of tests to verify alps release tunables namely alps_release_interval
+    and alps_release_jitter
+    """
+
+    def setUp(self):
+        machine = self.du.get_platform()
+        if not machine.startswith('cray') or machine == "craysim":
+            self.skipTest("Test suite only meant to run on a Cray")
+        TestFunctional.setUp(self)
+
+    def get_epoch(self, msg):
+        a = time.strptime(msg[:19], "%m/%d/%Y %H:%M:%S")
+        return int(time.mktime(a))
+
+    def test_alps_release_interval(self):
+        """
+        Set alps_release_interval to a higher value and then notice that
+        subsequest reservation cancellation requests are made atleast
+        after the set interval.
+        """
+        ari = 4.298
+        self.mom.add_config({'$alps_release_interval': ari})
+
+        # submit a job and then delete it after it starts running
+        start_time = time.time()
+        j1 = Job(TEST_USER)
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+        self.server.delete(jid1)
+
+        # Look for a message that confirms that reservation is deleted
+        self.mom.log_match("%s;ALPS reservation cancelled" % jid1,
+                           starttime=start_time, max_attempts=5,
+                           interval=2)
+        # Now that we know that reservation is cleared we should
+        # check for time difference between each cancellation request
+        out = self.mom.log_match("%s;Canceling ALPS reservation *" % jid1,
+                                 n='ALL', max_attempts=5, regexp=True,
+                                 allmatch=True)
+
+        # We found something, Let's first check there are atleast 2 such
+        # log messages, If not then that means reservation was cancelled
+        # in the first attempt itself, at that point right thing to do is
+        # to either run it again or find out a way to delay the reservation
+        # cancellation at ALPS level itself.
+        if len(out) >= 2:
+            nline_prev, line_prev = out[0]
+            time_prev = self.get_epoch(line_prev)
+            for data in out[1:]:
+                nline, line_current = data
+                time_current = self.get_epoch(line_current)
+                self.assertGreaterEqual(time_current - time_prev,
+                                        math.ceil(ari),
+                                        msg="alps_release_interval not working"
+                                        )
+                time_prev = time_current
+
+        else:
+            self.skipTest("Reservation cancelled without retry, Try again!")
```

Comment:
Instead of skipping this test, maybe put the meat of it into a function and try running it a few times before skipping or failing the test.

Reply:
I do something similar to what you are saying while testing Jitter. That I do because Jitter is random in nature and I can never be certain of how long it will sleep. In this test there is minimal jitter (120 miliseconds) so it is more certain in nature. I believe ALPS cancels in first request itself then it will get cancelled in every other try as well. I'd like to retain it as is but If you still think it makes sense to retry then I'll do that.